### ISSUE TITLE
Stream sound as soon as possible

### ISSUE THREAD

**Issue by tdowgielewicz** [polygraphene/ALVR#403]
Hi, I believe there is a way to increase imersion. I dont know how ALVR is working now. However it seams that you are sending audio stream with video stream with same latency. For imersion more noticable are lags from audio source than from image.  If you could separate this two streams and send audio sooner(ASAP) it would have very very good impact on gameplay.

**Comment by simphax** [comment]
I think its the opposite actually. Our brains are used to compensate for delayed sound (objects far away), while sound arriving earlier than the visuals never happen in real life.

**Comment by tdowgielewicz** [comment]
Many thanks for your answer. 

Your point is true. But,  it's based on the assumption that we live in real physical world.  However in this (brave)  new VR world. We have huge latency. And this latency breaks all equations. Imo only way to find out is to test it.

**Comment by simphax** [comment]
I would say something is wrong with the setup if there is “huge latency”. But these are all my opinions :) I definitely agree with testing

**Comment by jackfruh** [comment]
A good way to test this might be to play in the same room as your PC and not stream audio to the headset but instead use the speakers on your computer - then you'd hear the biggest difference and could see if this would make a difference in games you're interested in.

I do agree with Simphax, we should all want to reduce all latency as much as possible.

I am on the fence as to the benefit of lower latency for audio vs video - there are a lot of content types (movie playback, cut scenes, any game with a character that talks) where Audio and video really need to be in sync.

**Comment by tdowgielewicz** [comment]
Brilliant idea, i will try on weekend. I will also try to conect deluxe audio strap to pc by long jack cable.

**Comment by tdowgielewicz** [comment]
As @jackfruh  suggested i have been testing ALVR with instant audio

Setup:
i7 4790k + gtx970
ALVR 2.4.0. alpha5 latency ~70ms
Oculus Quest with Deluxe Audio Strap Mod https://imgur.com/a/kqLNhmb


Tests:
Audio Streamed By ALVR **[Latency]**
Audio Send Directly from PC **[Instant]**

Tested Apps / Games
- The Lab: Bow and Arrow Tower Defence
- The forrest
- Bigscreen: Some Youtube vlog

My conclusions.

Watching vlog on bigscreen in fact was a little bit wired. I have noticed that something is wrong when observing the lips. However it was not that bad. I was able to enjoy watching movie in both Tests

Running the lab was much more different. I event don't need to run the lab, to tell the differnece. Try to do teleport in steamVR room. It simply feels right. 

In the lab arrow game also there was a noticeable immersion boost. While moving controllers and tensing the bow it also feels much more better in instant mode. While playing on latency audio i felt 'off' like it was not me playing. I dont feel it in instant mode.
When releasing arrow it also feels better than in latency version

And my favorite game the forest. A lot of game play is cuting the trees. And also this feels very right comparing to the latency mode. 
I was not able to play the forest on ALVR version. I needed to lunch vive and play directly. But now im considering selling the vive 

I didnt notice any discomfort caused by audio played to quick for image.

Apparently there are two important factors in audio delay. One is from the natural sync 
Another one is expectations. When you pres / release the button you expect something to happen. If at least sound is there it feels better.

My suggestion would be to  have an option for user to have movie mode and game mode with different audio latency. 


TLDR: Instant sound improves immersion in games a lot . 

PS.
I dont know if you are planing to implement this feature or not, however i know for myslelf i will be buying  wireless audio transmiter. There is no going back in games :)

**Comment by orangpelupa** [comment]
hiya, i also got audio lag with
Win 10 1803 (will upgrade to 1909 later)
intel i5 4460
GTX 1660 super
ALVR v8

But i have NO NOTICABLE audio lag with
ASUS rog 702zc
Ryzen 7 1700
RX580
ALVR v8


Lag are noticeable in
Windows sound setting (go click the volume mixer slider, the DING sound will come late)
Beat saber (the block slicing SFX will come late)


I have not tried Oculus Link (no free space on C:)
